x-pack/plugins/ml/server/plugin.ts

```diff
@@ -0,0 +1,170 @@
+          savedObject: {
+            all: [],
+            read: [],
+          },
+          ui: [],
+        },
+        description: i18n.translate('xpack.ml.feature.reserved.description', {
+          defaultMessage:
+            'To grant users access, you should also assign either the machine_learning_user or machine_learning_admin role.',
+        }),
+      },
+    });
+
+    // Can access via router's handler function 'context' parameter - context.ml.mlClient
+    const mlClient = coreSetup.elasticsearch.createClient('ml', {
+      plugins: [elasticsearchJsPlugin],
+    });
+
+    coreSetup.http.registerRouteHandlerContext('ml', (context, request) => {
+      return {
+        mlClient: mlClient.asScoped(request),
+      };
+    });
+
+    const routeInit = {
+      router: coreSetup.http.createRouter(),
+      getLicenseCheckResults: () => this.licenseCheckResults,
+    };
+
+    annotationRoutes({
+      ...routeInit,
+      securityPlugin: plugins.security,
+    });
+    calendars(routeInit);
+    dataFeedRoutes(routeInit);
+    dataFrameAnalyticsRoutes(routeInit);
+    dataRecognizer(routeInit);
+    dataVisualizerRoutes(routeInit);
+    fieldsService(routeInit);
+    fileDataVisualizerRoutes(routeInit);
+    filtersRoutes(routeInit);
+    indicesRoutes(routeInit);
+    jobAuditMessagesRoutes(routeInit);
+    jobRoutes(routeInit);
+    jobServiceRoutes(routeInit);
+    notificationRoutes(routeInit);
+    resultsServiceRoutes(routeInit);
+    jobValidationRoutes({ ...routeInit, version: this.version });
+    systemRoutes({
+      ...routeInit,
+      spacesPlugin: plugins.spaces,
+      cloud: plugins.cloud,
+    });
+    initMlServerLog({ log: this.log });
+    coreSetup.getStartServices().then(([core]) => {
+      makeMlUsageCollector(plugins.usageCollection, core.savedObjects);
+    });
+
+    plugins.licensing.license$.subscribe(async license => {
+      const { isEnabled: securityIsEnabled } = license.getFeature('security');
+      // @ts-ignore isAvailable is not read
+      const { isAvailable, isEnabled } = license.getFeature(this.pluginId);
+
+      this.licenseCheckResults = {
+        isActive: license.isActive,
+        // This `isAvailable` check for the ml plugin returns false for a basic license
+        // ML should be available on basic with reduced functionality (only file data visualizer)
+        // TODO: This will need to be updated in the second step of this cutover to NP.
+        isAvailable: isEnabled,
+        isEnabled,
+        isSecurityDisabled: securityIsEnabled === false,
+        type: license.type,
+      };
+
+      if (sampleLinksInitialized === false) {
+        sampleLinksInitialized = true;
+        // Add links to the Kibana sample data sets if ml is enabled
+        // and license is trial or platinum.
+        if (isEnabled === true && plugins.home) {
+          if (
+            this.licenseCheckResults.type &&
+            ['platinum', 'trial'].includes(this.licenseCheckResults.type)
```

Comment:
This should include `enterprise` too, in line with the list of  `VALID_FULL_LICENSE_MODES` in `server/lib/check_license/check_license.ts`. We should probably define that list of full license modes in e.g. `common/constants/license.ts`.

Reply:
Updated and moved to `constants/license.ts` in 609be5c4c218d1feed4b249158c438ce7c30f99d